### PR TITLE
Fallback to no override flow when missing in client override

### LINKED ISSUE #30765: fallback to the no override flow when the flow is missing in client override
See the parent issue https://github.com/keycloak/keycloak/issues/30706 for context.

### PR DESCRIPTION
**Pull request by lexcao**
Closes #30765

### CODE REVIEW

**Comment by ahus1** [comment]
@graziang - this PR is related to an issue you commented on, therefore pinging you here.

**Review by graziang — APPROVED** [review]
@lexcao Thanks! LGTM. @ahus1 let's wait for @mposolda's review before merging

File: server-spi-private/src/main/java/org/keycloak/models/utils/AuthenticationFlowResolver.java
Comment (mposolda) [inline server-spi-private/src/main/java/org/keycloak/models/utils/AuthenticationFlowResolver.java:42]: Minor: The `flow` in this logging message will be always null if I am looking correctly? Should the third argument of the `logger.warnf` rather be `clientFlow` variable?

File: server-spi-private/src/main/java/org/keycloak/models/utils/AuthenticationFlowResolver.java
Comment (mposolda) [inline server-spi-private/src/main/java/org/keycloak/models/utils/AuthenticationFlowResolver.java:56]: Same comment as above

**Review by mposolda — COMMENTED** [review]
@lexcao @graziang Thanks for the fix and review! Added two other minor inline comments regarding logging message. IMO this can be merged once this is addressed.

**Comment by lexcao** [comment]
Hi @mposolda 
Thanks, my mistake, just updated the code.

**Review by mposolda — APPROVED** [review]
@lexcao Thanks for the update!

**Pull request merged**
